src/inverted_index.c

```diff
@@ -1161,25 +1172,26 @@ static void IndexReader_Init(const IndexSpec *sp, IndexReader *ret, InvertedInde
   ret->br = NewBufferReader(&IR_CURRENT_BLOCK(ret).buf);
   ret->decoders = decoder;
   ret->decoderCtx = decoderCtx;
+  ret->filterCtx = *filterCtx;
   ret->isValidP = NULL;
-  ret->sp = sp;
+  ret->sctx = sctx;
   IR_SetAtEnd(ret, 0);
 }
 
-static IndexReader *NewIndexReaderGeneric(const IndexSpec *sp, InvertedIndex *idx,
+static IndexReader *NewIndexReaderGeneric(const RedisSearchCtx *sctx, InvertedIndex *idx,
                                           IndexDecoderProcs decoder, IndexDecoderCtx decoderCtx, int skipMulti,
-                                          RSIndexResult *record) {
+                                          RSIndexResult *record, const FieldFilterCtx* filterCtx) {
   IndexReader *ret = rm_malloc(sizeof(IndexReader));
-  IndexReader_Init(sp, ret, idx, decoder, decoderCtx, skipMulti, record);
+  IndexReader_Init(sctx, ret, idx, decoder, decoderCtx, skipMulti, record, filterCtx);
   return ret;
 }
 
-IndexReader *NewTermIndexReader(InvertedIndex *idx, IndexSpec *sp, t_fieldMask fieldMask,
+IndexReader *NewTermIndexReader(InvertedIndex *idx, const RedisSearchCtx *sctx, t_fieldMask fieldMask,
                                 RSQueryTerm *term, double weight) {
-  if (term && sp) {
+  if (term && sctx) {
```

Comment:
isn't there a case where we have search context but no spec?

Reply:
Outside the normal flow there are the coordinator and unit tests.
In the coordinator case I don't think we ever get to an inverted index since it uses a network result processor.
in the case of the unit tests seems like a no since they all passed.

Reply:
OK. Is there a case where `sctx` **is** NULL? If only in unit-tests, can we have a `sctx` in them?

Reply:
Checked, saw calls with null in:
`TEST_F(IndexTest, testWeight)`
`SpellCheck_GetScore`
Also it is used indirectly by the debug commands.
Don't think it makes sense to add contect in the spell check and in debug commands.